src/main/java/com/rackspace/jenkins_nodepool/NodePoolSlave.java

```diff
@@ -297,6 +318,29 @@ boolean isBuildComplete() {
         }
     }
 
+    /**
+     * A convenience routine to help determine if the builds are finished.
+     *
+     * @param builds a list of builds for this slave
+     * @return true if all the builds are finished, false otherwise
+     */
+    private boolean isAllBuildsFinished(RunList builds) {
+        LOG.log(INFO, "Testing if all builds are finished...");
+        for (Object obj : builds) {
+            final Run build = (Run) obj;
+
+            // DEBUG
+            LOG.log(FINE, String.format("Build Result: %s, result.isCompleteBuild(): %s",
+                    build.getResult(), (build.getResult() == null ? "null" : build.getResult().isCompleteBuild())));
+
+            if (build.getResult() == null || !build.getResult().isCompleteBuild()) {
```

Comment:
I think we should only use the null check here, as `isCompleteBuild()` will return `false` for builds that have been aborted, as they didn't complete.

Reply:
👍